src/probability/kernel/cond_cdf.lean

```diff
@@ -0,0 +1,969 @@
+  tendsto f at_top at_bot ∨ (∃ l, tendsto f at_top (𝓝 l)) :=
+@tendsto_of_monotone ι αᵒᵈ _ _ _ _ _ h_mono
+
+-- todo: move to data/real/ennreal
+lemma ennreal.of_real_cinfi (f : α → ℝ) [nonempty α] :
+  ennreal.of_real (⨅ i, f i) = ⨅ i, ennreal.of_real (f i) :=
+begin
+  by_cases hf : bdd_below (range f),
+  { exact monotone.map_cinfi_of_continuous_at ennreal.continuous_of_real.continuous_at
+      (λ i j hij, ennreal.of_real_le_of_real hij) hf, },
+  { symmetry,
+    rw [real.infi_of_not_bdd_below hf, ennreal.of_real_zero, ← ennreal.bot_eq_zero, infi_eq_bot],
+    obtain ⟨y, hy_mem, hy_neg⟩ := not_bdd_below_iff.mp hf 0,
+    obtain ⟨i, rfl⟩ := mem_range.mpr hy_mem,
+    refine λ x hx, ⟨i, _⟩,
+    rwa ennreal.of_real_of_nonpos hy_neg.le, },
+end
+
+-- todo: move to measure_theory/measurable_space
+/-- Monotone convergence for an infimum over a directed family and indexed by a countable type -/
+theorem lintegral_infi_directed_of_measurable {mα : measurable_space α} [countable β]
+  {f : β → α → ℝ≥0∞} {μ : measure α} (hμ : μ ≠ 0)
+  (hf : ∀ b, measurable (f b)) (hf_int : ∀ b, ∫⁻ a, f b a ∂μ ≠ ∞) (h_directed : directed (≥) f) :
+  ∫⁻ a, ⨅ b, f b a ∂μ = ⨅ b, ∫⁻ a, f b a ∂μ :=
+begin
+  casesI nonempty_encodable β,
+  casesI is_empty_or_nonempty β,
+  { simp only [with_top.cinfi_empty, lintegral_const],
+    rw [ennreal.top_mul, if_neg],
+    simp only [measure.measure_univ_eq_zero, hμ, not_false_iff], },
+  inhabit β,
+  have : ∀ a, (⨅ b, f b a) = (⨅ n, f (h_directed.sequence f n) a),
+  { refine λ a, le_antisymm (le_infi (λ n, infi_le _ _))
+      (le_infi (λ b, infi_le_of_le (encodable.encode b + 1) _)),
+    exact (h_directed.sequence_le b a), },
+  calc ∫⁻ a, ⨅ b, f b a ∂μ
+        = ∫⁻ a, ⨅ n, f (h_directed.sequence f n) a ∂μ : by simp only [this]
+    ... = ⨅ n, ∫⁻ a, f (h_directed.sequence f n) a ∂μ :
+      by { rw lintegral_infi (λ n, _) h_directed.sequence_anti,
+        { exact hf_int _, },
+        { exact hf _, }, }
+    ... = ⨅ b, ∫⁻ a, f b a ∂μ :
+    begin
+      refine le_antisymm (le_infi (λ b, _)) (le_infi (λ n, _)),
+      { exact infi_le_of_le (encodable.encode b + 1) (lintegral_mono $ h_directed.sequence_le b) },
+      { exact infi_le (λb, ∫⁻ a, f b a ∂μ) _ },
+    end
+end
+
+-- todo: move to measure_theory/pi_system
+lemma is_pi_system_Iic [semilattice_inf α] : @is_pi_system α (range Iic) :=
+by { rintros s ⟨us, rfl⟩ t ⟨ut, rfl⟩ _, rw [Iic_inter_Iic], exact ⟨us ⊓ ut, rfl⟩, }
+
+-- todo: move to measure_theory/pi_system
+lemma is_pi_system_Ici [semilattice_sup α] : @is_pi_system α (range Ici) :=
+by { rintros s ⟨us, rfl⟩ t ⟨ut, rfl⟩ _, rw [Ici_inter_Ici], exact ⟨us ⊔ ut, rfl⟩, }
+
+
+end aux_lemmas_to_be_moved
+
+namespace measure_theory.measure
+
+variables {α β : Type*} {mα : measurable_space α}
+
+include mα
+
+/-- Measure on `α` such that for a measurable set `s`, `ρ.Iic_snd r s = ρ (s ×ˢ Iic r)`. -/
+noncomputable
+def Iic_snd (ρ : measure (α × ℝ)) (r : ℚ) : measure α :=
```

Comment:
Of course in the applications you will restrict to rational `r`, but in the definition and most properties you could (should?) use a general real `r`, right?